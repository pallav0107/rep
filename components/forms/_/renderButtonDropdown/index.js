export { default as renderButtonDropdown } from './ButtonDropdown';
