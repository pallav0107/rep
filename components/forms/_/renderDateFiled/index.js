export { default as renderDateFiled } from './InputDate';