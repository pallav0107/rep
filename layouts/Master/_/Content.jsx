import styled from 'styled-components';

export const Content = styled.main`
  display: flex;
  flex: 1 0 auto;
`;
