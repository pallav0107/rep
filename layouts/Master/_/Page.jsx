import styled from 'styled-components';
import { Flex } from 'grid-styled';

export const Page = styled(Flex)`
  min-height: 100vh;
  flex: 1;
  flex-direction: column;
`;
