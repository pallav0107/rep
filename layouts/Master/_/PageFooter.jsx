import React from 'react';
import { Footer } from '@wdpui/react-gel';
import { Text } from '@wdpui/gel-typography';

export const PageFooter = props => (
  <Footer>
    <Text>
      Our site and your transactions are secure. You can read our{' '}
      <a href="">security information</a>.
      <br />
      test application
    </Text>
  </Footer>
);

export default PageFooter;
