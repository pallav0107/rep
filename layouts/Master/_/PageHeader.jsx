import React from 'react';
import { Header, Button } from '@wdpui/react-gel';

const PageHeader = () => (
  <Header
    logoLarge
    logoCenterXs
    leftArrow
    leftArrowAction={() => window.history.back()}
  >
    <Button
      styling="faint"
      soft
      label="Sign out"
      responsive={{ xs: 'small', sm: 'medium' }}
      onClick={() => alert('Sign Out Clicked')}
    />
  </Header>
);

export default PageHeader;
