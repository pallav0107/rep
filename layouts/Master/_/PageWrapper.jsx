import styled from 'styled-components';
import { Flex } from 'grid-styled';

export const PageWrapper = styled(Flex)`
  max-width: ${props => (props.maxwidth ? props.maxwidth : '1922px')};
  flex-direction: column;
  margin-left: auto;
  margin-right: auto;
`;
