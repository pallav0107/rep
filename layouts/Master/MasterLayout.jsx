import React from 'react';
import PropTypes from 'prop-types';
import { PageWrapper } from './_/PageWrapper';
import { Page } from './_/Page';
import { Container } from './_/Container';
import { Content } from './_/Content';
import { PageFooter } from './_/PageFooter';
import PageHeader from './_/PageHeader';

const MasterLayout = ({ children }) => (
  <PageWrapper>
    <Page>
      <PageHeader />
      <Container column flex="1 0 auto">
        <Content>{children}</Content>
      </Container>
      <PageFooter />
    </Page>
  </PageWrapper>
);

MasterLayout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default MasterLayout;
