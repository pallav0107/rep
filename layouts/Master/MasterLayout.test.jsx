import React from 'react';
import MasterLayout from './MasterLayout';
import { shallow } from 'enzyme';
import { PageWrapper } from './_/PageWrapper';
import { Page } from './_/Page';
import PageHeader from './_/PageHeader';
import { Container } from './_/Container';
import { Content } from './_/Content';
import { PageFooter } from './_/PageFooter';
import { withTheme } from '../../../utils/jest/TestUtils';
import { Text } from '@wdpui/gel-typography';
import "../../setupTests"

describe('Master Layout', () => {
  it('should render the master layout structure', () => {
    const component = withTheme(<MasterLayout>Test Child</MasterLayout>);
    const wrapper = shallow(component).dive();

    expect(wrapper.find(PageWrapper).exists()).toBe(true);
    expect(wrapper.find(Page).exists()).toBe(true);
    expect(wrapper.find(Container).exists()).toBe(true);
    expect(wrapper.find(Content).exists()).toBe(true);
    expect(wrapper.find(PageFooter).exists()).toBe(true);

    expect(wrapper).toMatchSnapshot();
  });

  it('should pass the children into the content area', () => {
    const childElement = <Text>I am the test child.</Text>;
    const component = withTheme(<MasterLayout>{childElement}</MasterLayout>);
    const wrapper = shallow(component).dive();

    expect(wrapper.find(Content).contains(childElement)).toBe(true);
  });
});
