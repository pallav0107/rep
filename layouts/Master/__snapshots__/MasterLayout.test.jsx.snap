// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Master Layout should render the master layout structure 1`] = `
<Styled(Flex)>
  <Styled(Flex)>
    <PageHeader />
    <Styled(Flex)
      column={true}
      flex="1 0 auto"
    >
      <styled.main>
        Test Child
      </styled.main>
    </Styled(Flex)>
    <Component />
  </Styled(Flex)>
</Styled(Flex)>
`;
